init-service/src/main/java/uk/ac/ox/ndph/mts/init_service/loader/Loader.java

```diff
@@ -43,9 +49,31 @@ public Loader(Trial trialConfig,
     @Override
     public void run(String... args) throws InterruptedException, IOException {
         // Give the other services some time to come online.
+        // We wait and query the discovery to see that our services are up and ready
         initProgressReporter.submitProgress(String.format(LoaderProgress.ENTRY_POINT.message(), delayStartInSeconds));
         Thread.sleep(delayStartInSeconds * 1000);
```

Comment:
Is this still required?

Reply:
As discussed. it is partially required. as in real life those services won't be available from the start (discovery also), so we give it a bit of grace (will reduce the 120 sec constant to a lower value)